src/Transports.AspNetCore/WebSockets/IOperationMessageProcessor.cs

```diff
@@ -0,0 +1,21 @@
+using GraphQL.Transport;
+
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// Processes a stream of messages received from a WebSockets client.
+/// <see cref="IDisposable.Dispose"/> must be called when the connection terminates.
+/// Methods defined within this interface need not be thread-safe.
+/// </summary>
+public interface IOperationMessageProcessor : IDisposable
+{
+    /// <summary>
+    /// Starts the connection initialization timer, if configured.
```

Comment:
connection initialization timer?

Reply:
Also I suspect there is some initialization context (at least connection object itself). How are you going to make a implementation?

Reply:
> connection initialization timer

Starts the timer that closes the connection if no ConnectionInit message is received after ConnectionInitWaitTimeout 

See: https://github.com/graphql-dotnet/server/blob/develop/src/Transports.AspNetCore/WebSockets/GraphQLWebSocketOptions.cs#L8..L12

> some initialization context

The initialization message does not need to create context at all -- its only purpose is to authenticate the connection (since browsers cannot send a HTTP Authorize header with the connection).  Per the protocol, any other message is refused until this "ConnectionInit" message is received and acknowledged.

> connection object itself

Yes, the `IOperationMessageProcessor` holds state.  Note that it is not possible for two simultaneous connections over a single WebSocket connection.  So it's basically just a state machine along with a list of active subscribers (see #806 ) and a few background timers.  See #810 for implementation.

Reply:
> Starts the timer that closes the connection if no ConnectionInit message is received after ConnectionInitWaitTimeout

I would write exactly that in xml comments.

>  Per the protocol, any other message is refused until this "ConnectionInit" message is received and acknowledged.

And may be that too.